cmd/validate.go

```diff
@@ -100,13 +103,21 @@ func validateMain(cmd *cobra.Command, args []string) exitCode {
 
 	rand.Seed(time.Now().UnixNano())
 	plugin.InitCache()
-	ctx := context.Background()
 
 	pw := progress.NewWriter()
 	pw.SetUpdateFrequency(50 * time.Millisecond)
 	pw.Style().Colors = progress.StyleColorsExample
 	go pw.Render()
 
+	// On Ctrl-C cancel the context to ensure plugin calls have a chance to cleanup.
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, syscall.SIGINT, syscall.SIGTERM)
+	ctx, cancel := context.WithCancel(context.Background())
+	go func() {
+		<-sigCh
+		cancel()
```

Comment:
Do we need to wait here to give stuff time to cleanup?

Reply:
Nope, we capture the signal. Execution halts because the context is cancelled, and we exit via normal control flow. So in most cases it gets
```
...^C
Testing -rs- /docker/containers/swarm_web_1/fs/proc/cpuinfo ... done! [3 in 364ms]
Testing l--- /docker/volumes/swarm_redis ... done! [3 in 1.177s]
25.00% [####..............] [1 in 1.866158s] ... Testing -rs- /docker/volumes/swarm_redis/appendonly.aof
Error validating Stream on /docker/volumes/swarm_redis/appendonly.aof, shutting down after Ctrl-C: context canceled
50.00% [#########.........] [2 in 1.966337s] ... Testing -rs- /docker/volumes/swarm_redis/appendonly.aof
Found 1 errors.
```

Reply:
Ah cool, that makes sense.